deepspeed/inference/engine.py

```diff
@@ -381,6 +394,15 @@ def load_module_recursive(module, prefix='', level=0):
 
         load_module_recursive(r_module)
 
+        embedding_weight = None
+
+        for n, p in r_module.named_parameters():
+            if "word_embeddings." in n or "embed_tokens." in n or "wte." in n:
+                embedding_weight = p
+        if embedding_weight is not None and hasattr(r_module, "lm_head") and hasattr(
+                r_module.lm_head, "weight") and r_module.lm_head.weight.is_meta:
```

Comment:
I don't understand this logic. 
1. Are we setting `r_module.lm_head.weight` to the last embedding weight in the module parameters? 
2. Is it expected that `r_module.lm_head.weight` will be overwritten multiple times?

Reply:
yes. KI also has such logic in https://github.com/microsoft/DeepSpeed/blob/master/deepspeed/module_inject/load_checkpoint.py#L245
it will not overwritten multiple time only happen once when r_module.lm_head.weight.is_meta = true, will be false after the replacement.

Reply:
@sywangyi, thanks for the explanation. 

I am not familiar with KI code, so I will leave it to the experts to comment further. But it seems to me that if the intention is to modify exactly one parameter then it would be clearer to extract that one parameter rather than looping over `model.parameters()`.